Building hpc-stack-nco branch in ubuntu 20.04 container with gcc 9 and mpich 3.3.2

**Describe the bug**
Using build_stack.sh fails to install hpc-stack-nco in ubuntu 20.04 container

**To Reproduce**
1. (terminal 1) git clone --branch feature/hpc-stack-nco https://github.com/NOAA-EMC/hpc-stack
2. (terminal 2) docker run --rm -it --name hpc-container noaaemc/ubuntu-base:v1
3. (terminal 1) cd hpc-stack
4. (terminal 1) docker cp . hpc-container:/home/builder/hpc-stack
5. (terminal 2) cd hpc-stack
6. (terminal 2) export HPC_MPI=mpich/3.3.2 && \
    ./build_stack.sh -p /home/builder/opt -c config/config_custom.sh -y config/stack_custom.yaml

Build fails with "[[: not found" error message.

Change all lib/build_*.sh from #!/bin/sh to #!/bin/bash

Repeat above steps and build fails with:

-- Could NOT find NetCDF_Fortran (missing: NetCDF_Fortran_LIBRARY NetCDF_Fortran_INCLUDE_DIR) 
CMake Error at src/flib/CMakeLists.txt:275 (message):
  Must have PnetCDF and/or NetCDF Fortran libraries
-- Configuring incomplete, errors occurred!
See also "/home/builder/hpc-stack/pkg/pio-2.5.1/build/CMakeFiles/CMakeOutput.log".
BUILD FAIL!  Lib: pio Error:1

**System:**
Ubuntu 20.04, gcc 9, mpich 3.3.2

Latest develop branch failed to build in ubuntu 20.04. Error messages are:
```
[ 39%] Building Fortran object sorc/ncep_post.fd/CMakeFiles/upp.dir/AVIATION.f.o
cd /home/builder/hpc-stack/pkg/upp-upp_v10.0.0/build/sorc/ncep_post.fd && /usr/bin/gfortran  -I/home/builder/opt/include_4 -I/home/builder/opt/include -I/usr/include/x86_64-linux-gnu/mpich  -g -fbacktrace -ffree-form -ffree-line-length-none -fconvert=big-endian -O3 -Jinclude   -fopenmp -c /home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/AVIATION.f -o CMakeFiles/upp.dir/AVIATION.f.o
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:106:15:

  106 |       allocate(QC_BL(im,jsta_2l:jend_2u,lm))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:202:15:

  202 |       allocate(hail_maxhailcast(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:231:15:

  231 |       allocate(shdmin(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:232:15:

  232 |       allocate(shdmax(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:233:15:

  233 |       allocate(lai(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:275:15:

  275 |       allocate(rainc_bucket1(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:277:15:

  277 |       allocate(rainnc_bucket1(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:279:15:

  279 |       allocate(pcp_bucket1(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:281:15:

  281 |       allocate(snow_bucket1(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/sorc/ncep_post.fd/ALLOCATE_ALL.f:283:15:

  283 |       allocate(graup_bucket1(im,jsta_2l:jend_2u))
      |               1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
make[2]: *** [sorc/ncep_post.fd/CMakeFiles/upp.dir/build.make:76: sorc/ncep_post.fd/CMakeFiles/upp.dir/ALLOCATE_ALL.f.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/build'
make[1]: *** [CMakeFiles/Makefile2:133: sorc/ncep_post.fd/CMakeFiles/upp.dir/all] Error 2
make[1]: Leaving directory '/home/builder/hpc-stack/pkg/upp-upp_v10.0.0/build'
make: *** [Makefile:130: all] Error 2
BUILD FAIL!  NCEPlib: upp Error:2
```

I did not write down which version passed my test on 11/20/2020, but I am guessing either 10a8431 or 941250a. I can find out which commit is the culprit if it is helpful.

--- a/config/stack_ufs_weather_ci.yaml
+++ b/config/stack_ufs_weather_ci.yaml
@@ -166,7 +166,7 @@ crtm:
   install_as: 2.3.0
 
 nceppost:
-  build: YES
+  build: NO
   version: dceca26
   install_as: dceca26
   openmp: ON

